experiments/experiments.go

```diff
@@ -4,11 +4,26 @@
 // It is intended for internal use by buildkite-agent only.
 package experiments
 
-var experiments = make(map[string]bool)
+var (
+	Available = map[string]struct{}{
+		"job-api":                       {},
+		"kubernetes-exec":               {},
+		"ansi-timestamps":               {},
+		"git-mirrors":                   {},
+		"flock-file-locks":              {},
+		"resolve-commit-after-checkout": {},
+		"descending-spawn-priority":     {},
+		"inbuilt-status-page":           {},
+	}
+
+	experiments = make(map[string]bool, len(Available))
+)
 
 // Enable a particular experiment in the agent.
-func Enable(key string) {
+func Enable(key string) (known bool) {
 	experiments[key] = true
+	_, known = Available[key] // is the experiment they've enabled one that we know of?
+	return known
 }
```

Comment:
I know a lot of people are vehemently against named return values. Is it necessary to trigger them in this case 😛

Reply:
![dent-you-either-die-a-hero-or-you-live-long-enough-to-see-yourself-become-the-villain](https://user-images.githubusercontent.com/15753101/228082271-148b36f1-38d4-4b2f-a8a8-fe80e4af9950.gif)

i'm a known named-return hater, buttttttttttt..... i think in this case the actual meaning of the return value might not be as clear as if it weren't named - if i were to see a function `func Enable(key string) bool`, i think my first guess of what the return value means would be "was it actually enabled or not". just this once, i'll ~enter the named return radicalisation pipeline~ allow it

Reply:
I'm on team "change the method name if the return value is ambiguous"